File: onyx-database-tests/src/main/java/entities/ValidateRequiredIDEntity.java
package entities;

import com.onyx.persistence.IManagedEntity;
import com.onyx.persistence.annotations.Attribute;
import com.onyx.persistence.annotations.Entity;
import com.onyx.persistence.annotations.Identifier;

/**
 * Created by timothy.osborn on 2/10/15.
 */
@Entity
public class ValidateRequiredIDEntity extends AbstractEntity implements IManagedEntity
{
    @Attribute
    @Identifier
    public String id;

    @Attribute(nullable = false)
    public String requiredString;

    @Attribute(size = 10)
    public String maxSizeString;

}

File: onyx-database-tests/src/main/kotlin/entities/ValidateRequiredIDEntity.kt
package entities

import com.onyx.persistence.IManagedEntity
import com.onyx.persistence.annotations.Attribute
import com.onyx.persistence.annotations.Entity
import com.onyx.persistence.annotations.Identifier

/**
 * Created by timothy.osborn on 2/10/15.
 */
@Entity
class ValidateRequiredIDEntity : AbstractEntity(), IManagedEntity {
    @Attribute
    @Identifier
    var id: String? = null

    @Attribute(nullable = false)
    var requiredString: String? = null

    @Attribute(size = 10)
    var maxSizeString: String? = null

}
